src/components/views/dialogs/security/AccessSecretStorageDialog.tsx

```diff
@@ -201,6 +208,50 @@ export default class AccessSecretStorageDialog extends React.PureComponent<IProp
         });
     };
 
+    private onResetAllClick = (ev: React.MouseEvent<HTMLAnchorElement>) => {
+        ev.preventDefault();
+        this.setState({resetting: true});
+    };
+
+    private onConfirmResetAllClick = async () => {
+        // Hide ourselves so the user can interact with the reset dialogs.
+        // We don't conclude the promise chain (onFinished) yet to avoid confusing
+        // any upstream code flows.
+        //
+        // Note: this will unmount us, so don't call `setState` or anything in the
+        // rest of this function.
+        Modal.toggleCurrentDialogVisibility();
+
+        // Force reset secret storage (which resets the key backup)
+        await accessSecretStorage(async () => {
+            // Now reset cross-signing so everything Just Works™ again.
+            const cli = MatrixClientPeg.get();
+            await cli.bootstrapCrossSigning({
+                authUploadDeviceSigningKeys: async (makeRequest) => {
+                    // XXX: Making this an import breaks the app.
```

Comment:
Hmm, it works fine in other components... 🤔

Reply:
This dialog is loaded async for some reason, which makes it special